plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-crud-controller.php

```diff
@@ -166,7 +166,17 @@ protected function save_object( $request, $creating = false ) {
 				return $object;
 			}
 
-			$object->save();
+			/**
+			 * Setting a migic property to indicate that this is a REST request.
+			 * We are doing this to ensure that unique SKU validation
+			 * is only triggered for REST requests where we are handling the exception.
+			 */
+			$object->is_rest_request = true;
```

Comment:
This throws a deprecation notice in PHP 8.2+:

```
PHP Deprecated:  Creation of dynamic property WC_Product_Simple::$is_rest_request is deprecated in /var/www/html/wp-content/plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-crud-controller.php on line 174
```

Reply:
Would it work to use [this function](https://github.com/woocommerce/woocommerce/blob/eabc599c5129493045a20d6fa9dfb0a867aff3eb/plugins/woocommerce/includes/class-woocommerce.php#L459) instead of setting a property?

Reply:
Works perfected added as part of https://github.com/woocommerce/woocommerce/pull/47476/commits/5b596d822472d79cecfb1449be3497b40fd0a2af